sdks/java/io/redis/src/test/java/org/apache/beam/sdk/io/redis/RedisIOTest.java

```diff
@@ -250,6 +260,102 @@ public void testWriteUsingDECRBY() {
     assertEquals(-1, count);
   }
 
+  @Test
+  public void testWriteStreams() {
+    ArrayList<String> streams = new ArrayList<String>();
+    for (int i = 0; i <= 10; i++) {
+      UUID uuid = UUID.randomUUID();
+      /* stream keys are uuids to ensure that test runs are idempotent */
+      streams.add(uuid.toString());
+    }
+    Map<String, String> fooValues = ImmutableMap.of("sensor-id", "1234", "temperature", "19.8");
+    Map<String, String> barValues = ImmutableMap.of("sensor-id", "9999", "temperature", "18.2");
+
+    List<KV<String, Map<String, String>>> fooData =
+        streams.stream().map(key -> KV.of(key, fooValues)).collect(Collectors.toList());
+
+    List<KV<String, Map<String, String>>> barData =
+        streams.stream().map(key -> KV.of(key, barValues)).collect(Collectors.toList());
+
+    List<KV<String, Map<String, String>>> allData =
+        Stream.of(fooData, barData).flatMap(Collection::stream).collect(Collectors.toList());
+
+    PCollection<KV<String, Map<String, String>>> write =
+        p.apply(
+            Create.of(allData)
+                .withCoder(
+                    KvCoder.of(
+                        StringUtf8Coder.of(),
+                        MapCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of()))));
+    write.apply(RedisIO.writeStreams().withEndpoint(REDIS_HOST, port));
+    p.run();
+
+    for (String stream : streams) {
```

Comment:
Please use matchers to express this in a more intuitive way, e.g.

```
    Set<String> streams = ImmutableSet.copyOf(transform(allData, KV::getKey)); // if using above
    for (String stream : streams) {
      List<StreamEntry> results = client.xrange(stream, null, null, Integer.MAX_VALUE);
      assertEquals(2, results.size());
      assertThat(transform(results, StreamEntry::getFields), hasItems(fooValues, barValues));
    }
```

Note, this uses a few static imports:
```
import static java.util.stream.Collectors.toList;
import static org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.Lists.transform;
import static org.hamcrest.CoreMatchers.hasItems;
import static org.hamcrest.MatcherAssert.assertThat;
```

Reply:
Woof, yes, that's much more concise, thank you.